File: remove-element/remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i, k = 0, 0
        
        while k < len(nums):
            if nums[k] != val:
                nums[i] = nums[k]
                i += 1
            k += 1
        
        return i

File: remove-element/remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    var i = 0;
    var len = nums.length;
    
    while (i < len) {
        if (nums[i] === val) {
            // remove number from list
            nums.splice(i, 1);
            // adjust list length
            // (could also set len = nums.length)
            len--;
        } else {
            // only increment index if
            // number is not a match
            i++;
        }
    }
    
    return len;
};